namui-cli/src/build/build_server/start_build.rs

```diff
@@ -0,0 +1,177 @@
+    run_wasm_bindgen::{run_wasm_bindgen, RunWasmBindgenOption},
+};
+
+pub struct RebuildCallbackOption {
+    pub error_messages: Vec<ErrorMessage>,
+    pub result_path: Option<String>,
+}
+
+pub type RebuildCallback = fn(option: RebuildCallbackOption) -> ();
+
+pub struct StartBuildOption {
+    pub callback: RebuildCallback,
+    pub watch_dir: String,
+    pub bundle: Arc<RwLock<Bundle>>,
+    pub web_server: Arc<WebServer>,
+    pub manifest_path: String,
+}
+
+pub async fn start_build<'a>(option: StartBuildOption) {
+    let (thread_sender, mut thread_receiver) = tokio::sync::mpsc::channel::<DebouncedEvent>(32);
+
+    thread::spawn(move || {
+        let (watcher_sender, watcher_receiver) = std::sync::mpsc::channel::<DebouncedEvent>();
+
+        let mut watcher = watcher(watcher_sender, Duration::from_secs(1)).unwrap();
+        watcher
+            .watch(option.watch_dir, RecursiveMode::Recursive)
+            .unwrap();
+
+        loop {
+            match watcher_receiver.recv() {
+                Ok(event) => {
+                    let _ = block_on(thread_sender.send(event));
+                }
+                Err(error) => eprintln!("{:?}", error),
+            }
+        }
+    });
+
+    let mut should_rebuild = true;
+    loop {
+        match should_rebuild {
+            true => {
+                rebuild(
+                    option.callback,
+                    option
+                        .bundle
+                        .clone(),
+                    option
+                        .web_server
+                        .clone(),
+                    option
+                        .manifest_path
+                        .clone(),
+                )
+                .await;
+                should_rebuild = false;
+            }
+            false => 'await_file_change_event: loop {
+                match thread_receiver
+                    .recv()
+                    .await
+                {
+                    Some(event) => match event {
+                        DebouncedEvent::Create(_)
+                        | DebouncedEvent::Remove(_)
+                        | DebouncedEvent::Rename(_, _)
+                        | DebouncedEvent::Write(_) => {
+                            should_rebuild = true;
+                            break 'await_file_change_event;
+                        }
+                        _ => (),
+                    },
+                    _ => (),
+                };
+            },
+        }
+
+        'clear_file_change_events: loop {
+            match thread_receiver.try_recv() {
+                Ok(event) => match event {
+                    DebouncedEvent::Create(_)
+                    | DebouncedEvent::Remove(_)
+                    | DebouncedEvent::Rename(_, _)
+                    | DebouncedEvent::Write(_) => should_rebuild = true,
+                    _ => (),
+                },
+                Err(_) => break 'clear_file_change_events,
+            }
+        }
+    }
```

Comment:
1. Rebuild If `should_rebuild` is true. If not, wait for file changes
2. If there was file changes while rebuilding, Clear events and make `should_rebuild` true.